package.json

```diff
@@ -63,19 +62,29 @@
     "eslint-plugin-mocha": "6.x",
     "eslint-plugin-react": "7.x",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "history": "^5.3.0",
     "history-v4": "npm:history@^4.9",
-    "history-v5": "npm:history@^5.0",
     "jest": "^29.0.3",
     "jest-environment-jsdom": "^29.0.3",
     "jest-fetch-mock": "^3.0.3",
-    "prettier-eslint-cli": "^5.x",
+    "prettier-eslint-cli": "^7.1.0",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
     "ts-node": "^10.9.1",
     "typescript": "^4.8.3"
   },
   "peerDependencies": {
+    "@types/history": "^4.9",
+    "history": "^4.9 || ^5.0.0",
     "react": "^16.13.1"
   },
+  "peerDependenciesMeta": {
+    "history": {
+      "optional": true
+    },
+    "@types/history": {
+      "optional": true
+    }
+  },
```

Comment:
`history` is optional because you don't need it unless you use `useCourseSearch`. Nothing in the library imports history, we just use it for typings.

`@types/history` is optional because (a) if you're using history v5, you don't need it; (b) if you're not using typescript, you don't need it.